tests/test_query_parsing.py

```diff
@@ -252,60 +257,60 @@ async def test_correct_approximate_query(self):
     register_inference_services(aidb_engine, data_dir)
 
     config = aidb_engine._config
-
-    # approx aggregation
-    query_str1 = '''
+    
+    queries = {
+    # approx aggregation as a subquery
+    "test_query_0": {
+        QUERY_STR: '''
                  SELECT AVG(x_min) FROM  objects00
                  WHERE frame > (SELECT AVG(frame) FROM blobs_00)
                  ERROR_TARGET 10% CONFIDENCE 95%;
-                 '''
-
-    normalized_query_str1 = ("SELECT AVG(objects00.x_min) "
+                 ''',
+        NORMALIZED_QUERY_STR: ("SELECT AVG(objects00.x_min) "
                              "FROM objects00 "
-                             "WHERE objects00.frame > (SELECT AVG(frame) FROM blobs_00) ERROR_TARGET 10% CONFIDENCE 95%")
-
-    correct_fp = [["blobs_00.frame > (SELECT AVG(frame) FROM blobs_00)"]]
-
-    correct_service = [{}]
-    correct_tables = [{}]
-    self._test_query(query_str1, config, normalized_query_str1, correct_fp, correct_service, correct_tables, 2)
-
-
-    # approx select as a subquery
-    query_str2 = '''
+                             "WHERE objects00.frame > (SELECT AVG(frame) FROM blobs_00) ERROR_TARGET 10% CONFIDENCE 95%"),
+ 
+        CORRECT_FP : [["blobs_00.frame > (SELECT AVG(frame) FROM blobs_00)"]],
+        
+        # filter predicates connected by OR are in same set
+        CORRECT_SERVICE: [{}],
+        CORRECT_TABLES: [{}],
+        NUM_QUERY: 2
```

Comment:
Why is this hard coded?

Reply:
This is used to check the correctness of parsing results. The hard coded parts are desired results.

Reply:
What does NUM_QUERY: 2 mean?

Reply:
The query contain a subquery, so there are two queries in total. Or the number of 'SELECT' in the query sentence.

Reply:
This should probably be renamed to something like NUMBER_OF_SELECT_CLAUSES or something shorter

Reply:
Okay sure, renaming it